test/shared/TestConnection.cs

```diff
@@ -108,16 +87,14 @@ public void Dispose()
                 offset += count;
             }
 
-            Assert.Equal(expected, new String(actual, 0, offset));
+            Assert.Equal(expected, new string(actual, 0, offset));
         }
 
         public async Task ReceiveEnd(params string[] lines)
         {
             await Receive(lines);
-            var ch = new char[128];
-            var count = await _reader.ReadAsync(ch, 0, 128).TimeoutAfter(TimeSpan.FromMinutes(1));
-            var text = new string(ch, 0, count);
-            Assert.Equal("", text);
+            _socket.Shutdown(SocketShutdown.Both);
+            await WaitForConnectionClose();
```

Comment:
Why is this better than calling `_reader.ReadAsync` and checking for the 0 count? The nice thing about the previous version is the assertion failure while show what data was received given unexpected data. It also has a timeout.